WalletWasabi.Fluent/ViewModels/Wallets/Home/History/Details/CoinJoinDetailsViewModel.cs

```diff
@@ -55,6 +50,8 @@ private void Update()
 			Confirmations = transaction.Confirmations;
 			IsConfirmed = Confirmations > 0;
 			TransactionId = transaction.Id;
+			ConfirmationTime = _wallet.Transactions.TryEstimateConfirmationTime(transaction.Id);
```

Comment:
This would be correct if it would work as I was expecting to work. But no.

`WalletTransactionModel.Cache` will only trigger if we add or remove from it. So I had to re-introduce my ugly events to work this around.